src/SprykerSdk/Zed/AppSdk/Communication/Console/AppConfigurationCreateConsole.php

```diff
@@ -0,0 +1,595 @@
+    {
+        if ($this->properties[$propertyName]['type'] === 'Integer' && !is_numeric($item)) {
+            if ($this->askChoiceQuestion($input, $output, $this->getTypeSwitchQuestion(), $this->getConfirmationOptions(), 0) == 'Yes') {
+                $this->setPropertyType(
+                    $propertyName,
+                    $this->askChoiceQuestion($input, $output, $this->getWidgetTypeQuestion(), $this->getWidgetTextAndRadioTypeOptions(), 0),
+                );
+            }
+
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getTypeSwitchQuestion(): string
+    {
+        return 'You entered string value while int expected. Do you want to switch type?';
+    }
+
+    /**
+     * @param string $propertyName
+     * @param string $type
+     *
+     * @return void
+     */
+    protected function setTextTypeItem(string $propertyName, string $type): void
+    {
+        $this->properties[$propertyName]['widget']['id'] = $type;
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return void
+     */
+    protected function getFieldsetInput(InputInterface $input, OutputInterface $output): void
+    {
+        if ($this->askChoiceQuestion($input, $output, $this->getGroupConfirmationQuestion(), $this->getConfirmationOptions(), 0) == 'Yes') {
+            $fieldsetOptions = array_keys($this->properties);
+
+            do {
+                $groupIndex = $this->setGroupName($this->askTextQuestion($input, $output, $this->getGroupNameQuestion()));
+
+                $fieldsetOptions = array_diff($fieldsetOptions, $this->setGroupFields(
+                    $groupIndex,
+                    $this->askMultipleChoiceQuestion($input, $output, $this->getGroupSelectFieldsQuestion(), $fieldsetOptions, ''),
+                ));
+            } while ($fieldsetOptions && $this->askChoiceQuestion($input, $output, $this->getAddMoreGroupConfigurationQuestion(), $this->getConfirmationOptions(), 0) == 'Yes');
+        }
+    }
+
+    /**
+     * @return string
+     */
+    protected function getGroupConfirmationQuestion(): string
+    {
+        return 'Do you want to group the configurations?';
+    }
+
+    /**
+     * @return string
+     */
+    protected function getGroupNameQuestion(): string
+    {
+        return 'Please enter a group name: ';
+    }
+
+    /**
+     * @return string
+     */
+    protected function getGroupSelectFieldsQuestion(): string
+    {
+        return 'Please select all fields that should be in this group';
+    }
+
+    /**
+     * @return string
+     */
+    protected function getAddMoreGroupConfigurationQuestion(): string
+    {
+        return 'Do you want to add more group configurations?';
```

Comment:
Can you remove methods that return these kinds of strings? It's hard to read your question methods and you add too many methods only for returning a string. 

If you need to use a string more than once you should create a constant rather than having a method for it.

Reply:
I removed maximum additional methods.